File: sukjoo/Programmers/Level_2/Lv2_카펫.java
public class Lv2_카펫 {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int [2];
        int a = (brown - 4)/2;
        for(int i = 0 ; i < a; i++){
            if(i*(a-i) == yellow){
                answer[0] = i+2;
                answer[1] = a-i+2;
            }
        }
        
        
        return answer;
    }
}

File: chaerim/Programmers/Level_2/Lv2_카펫.py
def solution(brown, yellow):
    answer = []
    found = False

    for x in range(1, int(brown/2)+1):
        if found:
            break
        for y in range(1, int(brown/2)+1):
            if 2*(x+y)-4 == brown:
                if brown+yellow == x*y:
                    answer.append(x)
                    answer.append(y)
                    found = True

    answer.sort(reverse=True)

    return answer